### ISSUE TITLE
[Question] Docker Swarm config

### ISSUE THREAD

**Issue by irux** [puppetlabs/puppetlabs-docker#638]
Hello everybody,

I was searching a module for docker swarm and I find this beautiful repo. I am not super experienced with Puppet but I would like to make a little poc with docker swarm.

I would like to know if someone knows how to do something like here : #39 . That is what I am searching for.

Thank you

**Comment by oschusler** [comment]
@irux PR #651 was just merged and basically contains what you are looking for. You can now access the docker swarm join token as a fact on the manager node. This enables you to export the docker::swarm resource for the worker nodes, on the manager node:

```
node 'manager' {
  docker::swarm {'cluster_manager':
    init           => true,
    advertise_addr => "${::ipaddress}",
    listen_addr    => "${::ipaddress}",
    require        => Class['docker'],
  }

  @@docker::swarm {'cluster_worker':
    join           => true,
    manager_ip     => "${::ipaddress}",
    token          => "${worker_join_token}",
    tag            => "cluster_join_command",
    require        => Class['docker'],
  }
}

node 'worker' {
  Docker::Swarm<<| tag == 'cluster_join_command' |>> {
    advertise_addr => "${::ipaddress}",
    listen_addr    => "${::ipaddress}",
  }
}
```

**Comment by chelnak** [comment]
Closing as this seems to be resolved by #651 :smile:
